lib/util.go

```diff
@@ -16,3 +18,16 @@ func GenerateUUID() (string, error) {
 	}
 	return hex.EncodeToString(uid[:]), nil
 }
+
+// NormalizeEmail returns the email address and strips all other text from the
+// input, e.g. "Barry Gibbs <bg@example.com>" becomes "bg@example.com".
```

Comment:
The whole thing started form the idea of validating emails. There are a bunch of unreliable regexs out there but there's also this library method. So, I decided to us the library instead.

The other part is that the challenge-response registration will be driven by someone's code. It might be our code in mysky but it might also be something else. So, that code might use the full email formatting shown in the comment. Then the validation would go through (i.e. it won't error out) but the email we pass to the DB and to the mail sender will be the full string. So, the user won't be able to login as expected. So I decided to normalize the email. And Matt requested that it happens on deserialization, so we can't forget it.

What we can do instead is validate the email by parsing it with the same library function and if the input doesn't match the address output we can throw an error. This should take care of the issue in a simpler way. But we'll still have the "how do we make sure we never forget to do this?" issue.

Reply:
Resolved by a more standard approach to validating emails.